### PR TITLE
Improve the ingress list and detail card

### PR DESCRIPTION
**Pull request by kairen**
The commit improved the Ingress list and detail card, but I think the rules field need to discuss.

The following link shows preview of this change:
https://drive.google.com/drive/folders/0B75zg6bt8JyKV2pTV1dobG1wNFU?usp=sharing

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Rules section should not be visible or content should be filled with `-` if list size is 0.

**Comment by kairen** [comment]
@maciaszczykm I fixed, the preview as below:

File: src/app/frontend/ingress/detail/info.html [hunk 1]
```diff
@@ -35,12 +35,13 @@
     </kd-info-card-entry>
   </kd-info-card-section>
   <kd-info-card-section name="[[Rules|Resource rules spec for the ingress.]]">
-    <div ng-repeat="rule in ::$ctrl.ingress.spec.rules">
+    <div ng-if="::$ctrl.ingress.spec.rules.length" ng-repeat="rule in ::$ctrl.ingress.spec.rules">
```
Comment (floreks) [inline src/app/frontend/ingress/detail/info.html]: This `ng-if` should be used directly at `kd-info-card-section` tag. If there are no rules to display just hide whole section.

Comment (kairen) [reply src/app/frontend/ingress/detail/info.html]: @floreks I fixed this tag by new commit, thank you for your instruction and advice!!

**Comment by kairen** [comment]
@floreks I fixed this tag by new commit, thank you for your instruction and advice!!

**Comment by kairen** [comment]
Hi @floreks, @maciaszczykm Could help me to review these changes?

File: src/app/frontend/ingress/list/card.html [hunk 2]
```diff
@@ -47,5 +47,13 @@
       </div>
       <div ng-if="::!$ctrl.ingress.objectMeta.creationTimestamp">-</div>
     </kd-resource-card-column>
+    <kd-resource-card-column class="kd-row-layout-column kd-icon-column">
+      <kd-resource-card-menu>
+        <kd-resource-card-delete-menu-item resource-kind-name="Ingress">
```
Comment (floreks) [inline src/app/frontend/ingress/list/card.html]: This is a display name. Should be localized.

File: src/app/frontend/ingress/list/card.html [hunk 3]
```diff
@@ -47,5 +47,13 @@
...
+        </kd-resource-card-delete-menu-item>
+        <kd-resource-card-edit-menu-item resource-kind-name="Ingress">
```
Comment (floreks) [inline src/app/frontend/ingress/list/card.html]: Same here.

File: src/app/frontend/ingress/list/card.html (hunk 2 above)
Comment (kairen) [reply src/app/frontend/ingress/list/card.html]: @floreks Okay, I will fix this issue in next commit.

**Comment by kairen** [comment]
Hi @floreks, @maciaszczykm I added the ingress resource localizable text.

File: src/app/frontend/ingress/detail/info.html [hunk 4]
```diff
@@ -30,5 +30,18 @@
...
+    <kd-info-card-entry title="[[Creation time|Resource info details section creation time entry.]]">
+      {{::$ctrl.ingress.objectMeta.creationTimestamp | date}}
+    </kd-info-card-entry>
+  </kd-info-card-section>
+  <kd-info-card-section ng-if="::$ctrl.ingress.spec.rules.length"
+                        name="[[Rules|Resource rules spec for the ingress.]]">
+    <div ng-repeat="rule in ::$ctrl.ingress.spec.rules">
+      <kd-info-card-entry title="{{::rule.host}}">
+        <div ng-repeat="http in ::rule.http.paths">
+          {{::http.path}} => {{::http.backend.serviceName}}:{{::http.backend.servicePort}}
```
Comment (floreks) [inline src/app/frontend/ingress/detail/info.html]: I have checked how it looks in the UI and I think it should be refactored. It is not clear what is shown before the`'=>` and after it.

I'd move it to a new list or Rules with 2 columns.

Comment (maciaszczykm) [reply src/app/frontend/ingress/detail/info.html]: Definitely, we need a separate card for it IMO.

Comment (kairen) [reply src/app/frontend/ingress/detail/info.html]: @maciaszczykm Yes, you are right! I can try to implement it.

**Pull request closed without merging**